File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];
        int index = 0;

        while(index < nums.length){
            int product = 1; 
        
            for(int x = 0; x < nums.length; x++){
                if(x == index){
                    continue;
                }
                product = product * nums[x];
            }
            res[index] = product;
            index++;
        }
    return res;
    }
}  

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = nums[0]
        curr = 0

        for num in nums:
            if curr+num < 0:
                curr = 0
            else:
                curr+=num
                res = max(res, curr)

        return res
                